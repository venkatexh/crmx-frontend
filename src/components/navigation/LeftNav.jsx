import { Link } from "react-router-dom";
import "../../sass/components/navigation.component.scss";
import { useDispatch, useSelector } from "react-redux";
import { changeWindow } from "../../redux/actions/account/changeWindow";

const LeftNav = () => {
  const dispatch = useDispatch();
  const state = useSelector(({ accountWindow }) => ({ accountWindow }));
  const handleWindowChange = (window) => {
    dispatch(changeWindow(window));
  };

  const windows = ["dashboard", "contacts", "campaigns", "tags", "surveys"];

  const NavLink = ({ window }) => {
    return (
      <>
        <Link to={`/${window}`} className={"linkTag"}>
          <div
            className={`${
              state.accountWindow === window ? "navHighlighted" : ""
            } navLink`}
            onClick={() => handleWindowChange(window)}
          >
            <span className={"text"}>
              {window.charAt(0).toUpperCase() + window.slice(1)}
            </span>
          </div>
        </Link>
        {/*<hr className={"line"} />*/}
      </>
    );
  };
  return (
    <div className={"navigation"}>
      <div className={"logo"}>CRMX</div>
      {windows.map((window) => {
        return <NavLink window={window} />;
      })}
    </div>
  );
};
export default LeftNav;
