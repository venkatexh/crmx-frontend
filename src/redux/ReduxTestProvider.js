import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
