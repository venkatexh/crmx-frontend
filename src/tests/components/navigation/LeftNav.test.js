import { shallow } from "enzyme";
import React from "react";
import LeftNav from "../../../components/navigation/LeftNav";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("Left nav component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <LeftNav />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
