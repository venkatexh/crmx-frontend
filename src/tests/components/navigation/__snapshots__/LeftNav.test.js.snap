// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left nav component snapshot 1`] = `ShallowWrapper {}`;
